include/command.h

```diff
@@ -177,24 +176,16 @@ namespace detail {
 		reduction_id rid;
 	};
 
-	struct shutdown_data {};
-
-	struct sync_data {
-		uint64_t sync_id;
-	};
-
-	using command_data = std::variant<nop_data, horizon_data, execution_data, push_data, await_push_data, reduction_data, shutdown_data, sync_data>;
+	using command_data = std::variant<std::monostate, epoch_data, execution_data, push_data, await_push_data, reduction_data>;
 
 	/**
 	 * A command package is what is actually transferred between nodes.
 	 */
 	struct command_pkg {
-		command_id cid;
-		command_type cmd;
+		command_id cid{};
+		std::optional<task_id> tid{};
```

Comment:
We have three command types that provide a `tid`: `EPOCH`, `HORIZON` and `EXECUTION`. There are multiple places where we need the tid of a serialized task command without caring about which of the above it is, so having a common field reduces the amount of boilerplate in these places.

I do realize this is not exactly a beautiful API, since in principle it allows mismatches that have to be `assert`ed away. To get rid of the optional we could instead have a tree of variants in `command_data` like so:
```c++
struct task_data {
    task_id tid;
    std::variant<std::monostate, epoch_data, execution_data> data;
};
```

However we would still have an unenforced invariant between `command_pkg::cmd` and `command_pkg::data`, which must always refer to compatible variants.